Dockerfile

```diff
@@ -2,15 +2,27 @@ FROM eclipse-temurin:11-jdk as build
 
 WORKDIR /tmp
 RUN apt update -y && \
-    apt install -y ant git ca-certificates python3-sphinx && \
-    git clone https://github.com/kbase/jars
+    apt install -y git ca-certificates python3-sphinx
 
-COPY . /tmp/workspace_deluxe
+WORKDIR /tmp/workspace
 
-RUN cd workspace_deluxe && \
-    make docker_deps
+# dependencies take a while to D/L, so D/L & cache before the build so code changes don't cause
+# a new D/L
+# can't glob *gradle because of the .gradle dir
+COPY build.gradle gradlew settings.gradle /tmp/workspace/
+COPY gradle/ /tmp/workspace/gradle/
+RUN ./gradlew dependencies
+
+# Now build the code
+COPY workspace.spec /tmp/workspace/workspace.spec
+COPY deployment/ /tmp/workspace/deployment/
+COPY docshtml /tmp/workspace/docshtml/
+COPY docsource /tmp/workspace/docsource/
+COPY lib /tmp/workspace/lib/
+COPY src /tmp/workspace/src/
+COPY war /tmp/workspace/war/
```

Comment:
Already set `WORKDIR /tmp/workspace`, maybe we can simply `/tmp/workspace` as `.` when doing copy

Reply:
I would rather keep paths absolute and explicit when possible. `gradlew` needs to run in the project directory AIUI